contracts/modules/v2/Claims.sol

```diff
@@ -0,0 +1,381 @@
+   *                          cover wording doesn't enforce a proof of loss.
+   *  @param ipfsProofHash    The IPFS hash required for proof of loss. It is ignored if hasProof
+   *                          is false
+   */
+  function submitClaim(
+    uint24 coverId,
+    uint96 requestedAmount,
+    bool hasProof,
+    string calldata ipfsProofHash
+  ) external payable override onlyMember {
+    uint32 coverStart;
+    uint32 coverPeriod;
+    uint8 payoutAsset;
+    {
+      uint96 coverAmount;
+      (
+        /*uint24 productId*/,
+        payoutAsset,
+        coverAmount,
+        coverStart,
+        coverPeriod,
+        /*uint80 nxmPrice*/
+      ) = cover().covers(coverId);
+      require(requestedAmount <= coverAmount, "Cannot claim more than the covered amount");
+    }
+
+    {
+      address owner = cover().ownerOf(coverId);
+      claimants.push(owner);
+      cover().transferFrom(owner, address(this), coverId);
+    }
+
+    if (hasProof) {
+      emit ProofSubmitted(coverId, msg.sender, ipfsProofHash);
+    }
+
+    Claim memory claim = Claim(
+      0,
+      coverId,
+      requestedAmount,
+      payoutAsset,
+      false, // payoutRedeemed
+      false // coverRedeemed
+    );
+
+    uint nxmPrice = pool().getTokenPrice(payoutAsset);
+
+    // Calculate the expected in NXM using the NXM price at cover purchase time
+    uint expectedPayoutInNXM = requestedAmount * PRECISION / nxmPrice;
+
+    // Determine the total rewards that should be minted for the assessors based on cover period
+    uint totalReward = expectedPayoutInNXM * config.rewardRatio * coverPeriod / 365 days
+    / RATIO_BPS;
+
+    uint dynamicDeposit = max(config.maxRewardNXM ** PRECISION, totalReward * nxmPrice / PRECISION);
+    uint minDeposit = 1 ether * uint(config.minAssessmentDepositRatio) / RATIO_BPS;
+
+    // If dynamicDeposit falls below minDeposit use minDeposit instead
+    uint deposit = minDeposit > dynamicDeposit ? minDeposit : dynamicDeposit;
+
+    require(
+      msg.value == deposit,
+      "Assessment deposit different than the expected value"
+    );
+
+    uint assessmentId = assessment().startAssessment(totalReward, deposit);
+    claim.assessmentId = uint80(assessmentId);
+    claims.push(claim);
+  }
+
+  function redeemClaimPayout(uint104 claimId) external override {
+    Claim memory claim = claims[claimId];
+    (
+      IAssessment.Poll memory poll,
+      /*uint128 totalAssessmentReward*/,
+      uint assessmentDeposit
+    ) = assessment().assessments(claim.assessmentId);
+
+    require(poll.accepted > poll.denied, "The claim needs to be accepted");
+
+    (,,uint8 payoutCooldownDays) = assessment().config();
+    require(
+      block.timestamp >= poll.end + payoutCooldownDays * 1 days,
+      "The claim is in cooldown period"
+    );
+
+    require(
+      block.timestamp < poll.end +
+      payoutCooldownDays * 1 days +
+      config.payoutRedemptionPeriodDays * 1 days,
+      "The redemption period has expired"
+    );
```

Comment:
in this case, how do we send the deposit to the pool?

Reply:
this also means that you invalidate the whole cover if one didn't redeem on time, even for a partial claim

Reply:
As long as the cover is active or in grace period, yes.